assembly/plaintime.ts

```diff
@@ -14,33 +34,187 @@ export class PlainTime {
+      return new PlainTime(
+        hour,
+        minute,
+        second,
+        millisecond,
+        microsecond,
+        nanosecond
+      );
+    } else {
+      const dateTime = PlainDateTime.fromString(time);
+      return new PlainTime(
+        dateTime.hour,
+        dateTime.minute,
+        dateTime.second,
+        dateTime.millisecond,
+        dateTime.microsecond,
+        dateTime.nanosecond
+      );
+    }
+  }
+
+  @inline
+  static from<T = TimeLike>(time: T): PlainTime {
+    if (isString<T>()) {
+      // @ts-ignore: cast
+      return this.fromString(<string>time);
+    } else {
+      if (isReference<T>()) {
+        if (time instanceof PlainTime) {
+          return this.fromPlainTime(time);
+        } else if (time instanceof TimeLike) {
+          return this.fromTimeLike(time);
+        } else if (time instanceof PlainDateTime) {
+          return this.fromPlainDateTime(time);
+        }
+      }
+      throw new TypeError("invalid time type");
+    }
+  }
+
   constructor(
-    readonly hour: i32        = 0,
-    readonly minute: i32      = 0,
-    readonly second: i32      = 0,
+    readonly hour: i32 = 0,
+    readonly minute: i32 = 0,
+    readonly second: i32 = 0,
     readonly millisecond: i32 = 0,
     readonly microsecond: i32 = 0,
-    readonly nanosecond: i32  = 0
+    readonly nanosecond: i32 = 0
   ) {
-    if (!(
-      checkRange(hour,        0,  23) &&
-      checkRange(minute,      0,  59) &&
-      checkRange(second,      0,  59) &&
-      checkRange(millisecond, 0, 999) &&
-      checkRange(microsecond, 0, 999) &&
-      checkRange(nanosecond,  0, 999)
-    )) throw new RangeError("invalid plain time");
+    if (
+      !(
+        checkRange(hour, 0, 23) &&
+        checkRange(minute, 0, 59) &&
+        checkRange(second, 0, 59) &&
+        checkRange(millisecond, 0, 999) &&
+        checkRange(microsecond, 0, 999) &&
+        checkRange(nanosecond, 0, 999)
+      )
+    )
+      throw new RangeError("invalid plain time");
+  }
+
+  with(timeLike: TimeLike): PlainTime {
+    return new PlainTime(
+      coalesce(timeLike.hour, this.hour),
+      coalesce(timeLike.minute, this.minute),
+      coalesce(timeLike.second, this.second),
+      coalesce(timeLike.millisecond, this.millisecond),
+      coalesce(timeLike.microsecond, this.microsecond),
+      coalesce(timeLike.nanosecond, this.nanosecond)
+    );
+  }
+
+  toString(): string {
+    // 22:54:31
+    return (
+      toPaddedString(this.hour) +
+      ":" +
+      toPaddedString(this.minute) +
+      ":" +
+      toPaddedString(this.second) +
+      ((this.nanosecond | this.microsecond | this.millisecond) != 0
+        ? (
+            f64(this.nanosecond) / 1_000_000_000.0 +
+            f64(this.microsecond) / 1_000_000.0 +
+            f64(this.millisecond) / 1_000.0
+          )
+            .toString()
+            .substring(1)
+        : "")
+    );
+  }
+
+  toPlainDateTime(
+    dateLike: DateLike = { year: 0, month: 0, day: 0 }
```

Comment:
According to this:
https://github.com/tc39/proposal-temporal/blob/2f36d91332aeebc3a66082600c8eef57b751fbd5/polyfill/test/PlainDate/prototype/toPlainDateTime/time-undefined.js#L10

better use this approach:
```ts
toPlainDateTime(dateLike: DateLike | null = null): PlainDateTime {
  let year = 0, month = 0, day = 0;
  if (dateLike !== null) {
     year  = dateLike.year;
     month = dateLike.month;
     day   = dateLike.day;
  }
  return new PlainDateTime(
    year,
    month,
    day,
    this.hour,
    this.minute,
    this.second,
    this.millisecond,
    this.microsecond,
    this.nanosecond
  );
}
```

Reply:
Done 👍